sdk/cosmosdb/azure_data_cosmos/examples/cosmosdb_connect.rs

```diff
@@ -0,0 +1,45 @@
+use azure_data_cosmos::{CosmosClient, CosmosClientMethods, DatabaseClientMethods};
+use clap::Parser;
+
+/// A simple example to show connecting to a Cosmos Account and retrieving the properties of a database.
+#[derive(Parser)]
+pub struct Args {
+    /// The CosmosDB endpoint to connect to.
+    endpoint: String,
+
+    /// The database to fetch information for.
+    database: String,
+
+    /// An authentication key to use when connecting to the Cosmos DB account. If omitted, the connection will use Entra ID.
+    #[clap(long)]
+    #[cfg(feature = "key-auth")]
+    key: Option<String>,
+}
+
+#[tokio::main]
+pub async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let args = Args::parse();
+
+    let client = create_client(&args);
+
+    let db_client = client.database(&args.database);
+    let response = db_client.read(None).await?.deserialize_body().await?;
+    println!("{:?}", response);
+    Ok(())
+}
+
+#[cfg(feature = "key-auth")]
+fn create_client(args: &Args) -> CosmosClient {
+    if let Some(key) = args.key.as_ref() {
+        CosmosClient::with_key(&args.endpoint, key.clone(), None).unwrap()
```

Comment:
How married to shared key auth are you? I don't remember if we discussed it with your team, but did discuss with Storage about not support shared key auth at all. Or was it deemed too ubiquitous for Cosmos?

Reply:
I don't think we'll be able to remove it completely. Certain control-plane operations, namely managing Databases and Containers, can be performed using the Data Plane REST APIs (for example, see the [Create Database](https://learn.microsoft.com/en-us/rest/api/cosmos-db/create-a-database) REST API) **but only** when authenticating with a shared key (if you want to perform those operations using Entra ID, you need to use the Control Plane APIs).

As a result, it's difficult to justify removing shared key auth _entirely_. My hope here is that by putting it behind a feature, we can walk a bit of a line here of providing access to those APIs while still making the "standard" experience support Entra ID only.